mesonbuild/modules/_qt.py

```diff
@@ -648,3 +733,431 @@ def compile_translations(self, state: 'ModuleState', args: T.Tuple, kwargs: 'Com
+            cmd,
+            inputs,
+            outputs,
+            install=kwargs['install'],
+            install_dir=install_dir,
+            install_tag=install_tag,
+            description=f'Qml type registration for {target_name}',
+        )
+
+    @FeatureNew('qt.qml_module', '1.6')
+    @typed_kwargs(
+        'qt.qml_module',
+        #qml sources
+        KwargInfo('qml_sources', ContainerTypeInfo(list, (File, str, build.CustomTarget)), listify=True, default=[]),
+        KwargInfo('qml_singletons', ContainerTypeInfo(list, (File, str, build.CustomTarget)), listify=True, default=[]),
+        KwargInfo('qml_internals', ContainerTypeInfo(list, (File, str, build.CustomTarget)), listify=True, default=[]),
+        KwargInfo('resources_prefix', str, default='qt/qml'),
+
+        #qmldir generation
+        KwargInfo('imports', ContainerTypeInfo(list, (str)), default=[]),
+        KwargInfo('optional_imports', ContainerTypeInfo(list, (str)), default=[]),
+        KwargInfo('default_imports', ContainerTypeInfo(list, (str)), default=[]),
+        #match DEPENDENCIES argument from CMake, but dependencies keyword is already taken
+        KwargInfo('depends_imports', ContainerTypeInfo(list, (str)), default=[]),
+        KwargInfo('designer_supported', bool, default=False),
+
+        #for type registration, same arguments as moc
+        KwargInfo('moc_headers', ContainerTypeInfo(list, (File, str, build.CustomTarget)), listify=True, default=[]),
+        KwargInfo('moc_sources', ContainerTypeInfo(list, (File, str, build.CustomTarget)), listify=True, default=[]),
+        KwargInfo('include_directories', ContainerTypeInfo(list, (build.IncludeDirs, str)), listify=True, default=[]),
+        KwargInfo('namespace', str, default=''),
+        KwargInfo('typeinfo', str, default=''),
+
+        KwargInfo('moc_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+        KwargInfo('qmlcachegen_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+        KwargInfo('qmltyperegistrar_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+
+        KwargInfo('generate_qmldir', bool, default=True),
+        KwargInfo('generate_qmltype', bool, default=True),
+        KwargInfo('cachegen', bool, default=True),
+
+        KwargInfo('dependencies', ContainerTypeInfo(list, (Dependency, ExternalLibrary)), listify=True, default=[]),
+        INSTALL_DIR_KW,
+        INSTALL_KW,
+        KwargInfo('method', str, default='auto'),
+        KwargInfo('preserve_paths', bool, default=False),
+    )
+    def qml_module(self, state: ModuleState, args: T.Tuple[str, str, str], kwargs: 'QmlModuleKwArgs') -> ModuleReturnValue:
+
+        self._detect_tools(state, kwargs['method'])
+        if not self._support_qml_module:
+            raise MesonException('qt.qml_module is not suppported for this version of Qt')
+
+        #Major.Minor(.Patch)
+        version_re = re.compile('^(\\d+)\\.(\\d+)(\\.(\\d+))?$')
```

Comment:
You can simplify this a bit by using `r'string'` (officially r for raw, not r for regex, but we all know the truth, eh?)

Reply:
updated,

note I moved the module regexp definition outside the class as it was a bit convoluted and reused multiple times